lib/widgets.php

```diff
@@ -124,10 +132,12 @@ function gutenberg_get_legacy_widget_settings() {
 	global $wp_registered_widgets;
 	if ( ! empty( $wp_registered_widgets ) ) {
 		foreach ( $wp_registered_widgets as $widget_id => $widget_obj ) {
+			$block_widget_start = 'blocks-widget-';
 			if (
-				is_array( $widget_obj['callback'] ) &&
+				( is_array( $widget_obj['callback'] ) &&
 				isset( $widget_obj['callback'][0] ) &&
-				( $widget_obj['callback'][0] instanceof WP_Widget )
+				( $widget_obj['callback'][0] instanceof WP_Widget ) ) ||
+				strncmp( $widget_id, $block_widget_start, strlen( $block_widget_start ) ) === 0
```

Comment:
It's more common (and, I think, easier to understand) to see `substr` used to check that a string begins with another string.

```php
substr( $widget_id, 0, strlen( $block_widget_start ) ) === $block_widget_start
```

Reply:
Performance is not very relevant in this case. But, substring first unconditionally iterates on the string to copy parts of it and the only in the end the comparison is done. The method I'm using does not copy any character and stops the comparison when the first char difference is found. So I normally use this code because it is more performant. I added a comment to say that this logic is a starts with.

Reply:
Interesting! PHP needs a `starts_with` function 😛